vars/simplePipeline.groovy

```diff
@@ -45,7 +45,19 @@ def call(Map pipelineParams) {
                     }
                 }
             }
-        }
+            stage('Docker') {
+                when {
+                    expression {
+                        return (pipelineParams.dockerFile != '' && pipelineParams.dockerImageTag != '');
+                    }
+                }
+                steps {
+                    container('docker-in-docker') {
```

Comment:
Using the term docker-in-docker here is confusing, especially as we are using the approach described in the post you reference, https://jpetazzo.github.io/2015/09/03/do-not-use-docker-in-docker-for-ci/ - i.e. *not* using docker-in-docker

Reply:
Done